src/components/FeatureViews/Courses/ModalTextEditor.js

```diff
@@ -134,122 +185,309 @@ const ModalTextEditor = ({
-      const sentEmail = await sendEmail({
-        variables: {
-          subject: subject,
-          body: body,
-          userId: userId,
-          posterId: poster_id,
-        },
-      });
+  useEffect(() => {
+    if(buttonState === "edit") {
+      setBody(EditorState.createWithContent(convertFromRaw(textBody)))
     } else {
-      const createdSessionNote = await createSessionNote({
-        variables: {
-          subject: subject,
-          body: body,
-          user: poster_id,
-          sessionId: sessionId,
-          id: noteId,
+      setBody(EditorState.createEmpty())
+    }
+  }, [buttonState])
+
+  const QUERIES = {
+    ANNOUNCEMENTS: GET_ANNOUNCEMENTS,
+    COURSE_SESSIONS: GET_SESSION_NOTES,
+  };
+
+  const QUERY_VARAIBLES = {
+    ANNOUNCEMENTS: {
+      id: courseId.id,
+    },
+    COURSE_SESSIONS: {
+      sessionId,
+    },
+  };
+
+  const MUTATION_VARIABLES = {
+    ANNOUNCEMENTS: {
+      subject,
+      body: convertToRaw(body.getCurrentContent()),
+      id: announcementId,
+      userId: poster_id,
+      courseId: courseId.id,
+      shouldEmail: sendEmailCheckbox,
+    },
+    STUDENT_ENROLLMENT: {
+      subject: subject,
+      body: body.getCurrentContent(),
+      userId,
+      posterId: poster_id,
+    },
+    COURSE_SESSIONS: {
+      subject: subject,
+      body: convertToRaw(body.getCurrentContent()),
+      user: poster_id,
+      sessionId: sessionId,
+      id: noteId,
+    },
+  };
+
+  const [mutateTextEditor, createResults] = useMutation(MUTATIONS[origin], {
+    onCompleted: () => handleClose(false),
+    update: (cache, { data }) => {
+      const [newTextData] = Object.values(data[MUTATION_KEY[origin]]);
+      const cachedTextData = cache.readQuery({
+        query: QUERIES[origin],
+        variables: QUERY_VARAIBLES[origin],
+      })[QUERY_KEY[origin]];
+      let updatedTextData = [...cachedTextData];
+      const matchingIndex = updatedTextData.findIndex(
+        ({ id }) => id === newTextData.id
+      );
+      if (matchingIndex === -1) {
+        updatedTextData = [...cachedTextData, newTextData];
+      } else {
+        updatedTextData[matchingIndex] = newTextData;
+      }
+
+      cache.writeQuery({
+        data: {
+          [QUERY_KEY[origin]]: updatedTextData,
         },
+        query: QUERIES[origin],
+        variables: QUERY_VARAIBLES[origin],
       });
+    },
+  });
+
+  const handleClose = () => {
+    handleCloseForm(false);
+    setBody(EditorState.createEmpty());
+  };
+
+  const handleSubmit = async e => {
+    e.preventDefault();
+    const createTextData = await mutateTextEditor({
+      variables: MUTATION_VARIABLES[origin],
+    });
+    setBody(EditorState.createEmpty())
+  };
+
+  const toggleInlineStyles = e => {
+    e.preventDefault();
+    const style = e.currentTarget.getAttribute("data-style");
+    const { toggleInlineStyle, toggleBlockType } = RichUtils
+    const newState = {
+      [style]: toggleInlineStyle(body, style),
```

Comment:
As written, this is the only key of the object that is ever used; did you mean to use quotes instead of brackets?